compiler/testData/codegen/box/when/patternMatching/test1.kt

```diff
@@ -0,0 +1,37 @@
+// WITH_RUNTIME
```

Comment:
I see only codegen tests here, and this is quite bad. You should add, at least, some new parsing tests, because you modified the parser (ParsingTestGenerated), and some front-end diagnostics test, because you modified front-end analysis (DiagnosticsTestGenerated).